### PR TITLE
make "image" dependency optional

### PR DESCRIPTION
**Pull request by vthriller**
This reduces dependency list by about 4% (428 down from 447 on linux).

### CODE REVIEW

**Comment by jonathandturner** [comment]
Thanks for the PR! I was wondering about this, too. Would it be better to make image optional, or the whole binaryview optional?  Binaryview will probably grow to hold a few different things in it.

**Comment by vthriller** [comment]
> Would it be better to make image optional, or the whole binaryview optional?

TBH I'm not sure either. The answer would probably depend on the list of planned formats and whether it makes sense to support them in every situation (just like it doesn't make sense to use `clipboard` on remote servers). But even then, I'm not sure the image viewer in particular will end up being more than just a gimmick.

File: src/plugins/binaryview.rs [hunk 1]
```diff
@@ -347,6 +353,18 @@ pub fn view_contents(
     Ok(())
 }
 
+#[cfg(not(feature = "image"))]
+#[allow(unused)]
```
Comment (bjorn3) [inline src/plugins/binaryview.rs]: Why this? If it is unused, can you please remove it.

Comment (vthriller) [reply src/plugins/binaryview.rs]: It's not the function that's unused but its argument, `lores_mode`. I totally forgot for a moment that I can just rename it, will do so in a moment.

**Comment by jonathandturner** [comment]
@vthriller - agreed displaying the image might be something you don't often use, what I was considering is that we may want to support pulling in the image metadata (size, # megapixels, location, etc).  Give people a way to get to see the image if they want, but mostly I think people would use it to make a script to eg go through a directory and resize images.

**Comment by vthriller** [comment]
> what I was considering is that we may want to support pulling in the image metadata (size, # megapixels, location, etc)

Should it be shell's job to support this and other sorts of (meta)data though?

Now that I think of it I don't even find the idea of some universal and somewhat magical `binaryview` any good. I'm not going to dive into discussion about what it should and should not do and whether it's even going to be useful to a lot of people, but if it is, it definitely should be a standalone tool with its own dependencies, release cycle etc. It's shell that's developing here, after all, and its job is not to do everything by itself but to run and couple things that can do all the magic bits from reading EXIF/XMP to asking storage medium for its SMART status to whatever, even if outside world is not quite ready for structured pipes yet.

**Comment by jonathandturner** [comment]
@vthriller - that's exactly why it's a plugin and not part of the core shell. Eventually, the plugins will live separately and will have their own recycle cycle and dependencies. You'll be able to install them and have them, but they won't be required.

It's just early days so we haven't done that separation, yet.

**Comment by jonathandturner** [comment]
@vthriller - been doing a lot of thinking about this after reading your comments. I think there's an approach I want to go with where commands hook into the viewer. This will let us make all the separate binary views loadable as plugins and not be all in one binaryview.

I'm going to go ahead and close this and then focus on splitting up binaryview into separate parts.

**Pull request closed without merging**